### ISSUE TITLE
Unexpected -Wconstant-logical-operand warning when compiled with c2x/gnu2x

### ISSUE THREAD

**Issue by fel1x-developer** [llvm/llvm-project#80548]
```
#include <stdio.h>

int main() {
	if ((64 > 32) && (32 < 64))
		printf("%lu\n", sizeof(int));

	return 0;
}
```

```
$ clang -std=c2x main.c
main.c:4:36: warning: use of logical '&&' with constant operand [-Wconstant-logical-operand]
    4 |         if ((sizeof(int) > sizeof(short)) && (sizeof(int) < sizeof(long)))
      |                                           ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
main.c:4:36: note: use '&' for a bitwise operation
    4 |         if ((sizeof(int) > sizeof(short)) && (sizeof(int) < sizeof(long)))
      |                                           ^~
      |                                           &
main.c:4:36: note: remove constant to silence this warning
    4 |         if ((sizeof(int) > sizeof(short)) && (sizeof(int) < sizeof(long)))
      |                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

Build with `-std=c17` or `-std=gnu17` and below prints no error, but this emits warnings when built with `c2x` or `gnu2x`. Is this expected behaviour that reflects some changes in C23 or a bug?

**Comment by shafik** [comment]
This looks related to: https://github.com/llvm/llvm-project/issues/64356

CC @AaronBallman

**Comment by EugeneZelenko** [comment]
@fel1x-developer: Could you please elaborate?

**Comment by fel1x-developer** [comment]
> @fel1x-developer: Could you please elaborate?

This is duplicate of #64356, though it still remains in clang 17.0.6 (Darwin amd64/aarch64 on macOS Sonoma, installed through Homebrew) Is this issue resolved in Clang 18?
